### ISSUE TITLE
Client settings - Identifier  with IPv6(fe80::) can`t use the upstream dns settings.

### ISSUE THREAD

**Issue by whatagoodboy** [AdguardTeam/AdGuardHome#6872]
### Prerequisites

- [X] I have checked the [Wiki](https://github.com/AdguardTeam/AdGuardHome/wiki) and [Discussions](https://github.com/AdguardTeam/AdGuardHome/discussions/categories/q-a) and found no answer

- [X] I have searched other issues and found no duplicates

- [X] I want to report a bug and not [ask a question or ask for help](https://github.com/AdguardTeam/AdGuardHome/discussions/categories/q-a)

- [X] I have set up AdGuard Home correctly and [configured clients to use it](https://github.com/AdguardTeam/AdGuardHome/wiki/Clients). (Use the [Discussions](https://github.com/AdguardTeam/AdGuardHome/discussions/categories/q-a) for help with installing and configuring clients.)

### Platform (OS and CPU architecture)

Darwin (aka macOS), AMD64 (aka x86_64)

### Installation

Docker

### Setup

On one machine

### AdGuard Home version

v0.108.0-b.54

### Action

Client settings - Identifier  settings:
ipv4 can use the upstream dns, but ipv6 use the global settings upstream dns.

<img width="654" alt="settings" src="https://github.com/AdguardTeam/AdGuardHome/assets/6156891/e1f63445-6c09-4740-8de3-faa7e5791374">

### Expected result

ipv6 can use the Client settings upstream dns

### Actual result

ths is the result 

<img width="1209" alt="result" src="https://github.com/AdguardTeam/AdGuardHome/assets/6156891/fc3e65d4-aa6b-448b-8ea4-29c8b54d7da8">

**Comment by overwatch3560** [comment]
@whatagoodboy have you added any ipv6 dns upstream servers in the dns settings?

**Comment by whatagoodboy** [comment]
> @whatagoodboy have you added any ipv6 dns upstream servers in the dns settings?

yes, I added ipv6 dns (include  domains and  ip format), it still goes to the global dns.

**Comment by whatagoodboy** [comment]
> Are you using AGH DHCP Server, or another DHCP server?

@jslawler-gh I am using another dhcp server,  and the  dns set to AGH server.  My ipv4 clients settings goes to the correct dns. Only   ipv6  clients settings  goes wrong.

**Comment by whatagoodboy** [comment]
> Thank you for confirming. Which OS are you hosting your AGH server on?

@jslawler-gh Synology DSM7.2 -  Docker  , network=host

**Comment by whatagoodboy** [comment]
> @whatagoodboy thanks. I'll be as I won't be able to test that exact setup, I'll be shooting in the dark a bit here.
> 
> To determine where the issue is: Are you able to reproduce this outside of a containerised environment? Also, [can you check your IPv6 setup on the DSM](https://kb.synology.com/en-eu/DSM/help/DSM/AdminCenter/connection_network_lan?version=7)?

@jslawler-gh Thank you for your reply. ADG works fine with ipv6 in docker, as the second screenshot provided at the beginning shows, it is using global ipv6 dns (https://[2606:4700:4700::1001]:443/dns-query), and it already get the correct result. Can you try to have a test  use "Client settings" to customize upstream DNS? Does the customized upstream DNS was used correctly by client requests with ipv6 from local link fe80:: in your environment?
Sorry for poor English...

**Comment by rubeo-O** [comment]
I am also seeing this same issue where IPv6 identifiers cannot use custom upstream resolvers. I am running two instances of Adguard Home - one in an LXC and another in Docker.

**Comment by cashford8** [comment]
I also see this issue. My adguard is running as a snap on raspian OS. 
In addition to it not obeying the upstream DNS settings, the ipv6 client doesn't get any of the other custom settings when you untick "use global settings" in the settings tab - such as safe search options
The same client connected via ipv4 works as expected.

**Comment by spencertiedge** [comment]
+1 here — if I turn off IPV6 on my router, all client settings work as expected (blocked services, upstream DNS, etc.).

However, when client matches a IPV6 address (it shows correctly as such in the query log, matching both expected IPV6 address and client name), none of the custom settings applied at the client level are used.

**Comment by peter-m-calloway** [comment]
Same situation here! Everything is working correctly in IPv4, but specific client settings are not being applied in IPv6 (even when client matches are successful)
